cmp/form/dropdown/AsyncComboField.js

```diff
@@ -0,0 +1,85 @@
+/*
+ * This file belongs to Hoist, an application development toolkit
+ * developed by Extremely Heavy Industries (www.xh.io | info@xh.io)
+ *
+ * Copyright © 2018 Extremely Heavy Industries Inc.
+ */
+
+import {hoistComponent, elemFactory} from 'hoist/core';
+import {observable, setter} from 'hoist/mobx';
+import {Classes, suggest} from 'hoist/kit/blueprint';
+
+import {BaseDropdownField} from './BaseDropdownField';
+
+/**
+ * ComboBox Field, which populates its items via an async function (e.g. querying a remote server)
+ *
+ * @prop rest, see properties for HoistField
+ *
+ * @prop queryAsync, async function that receives (string query)
+ *          Should return a collection of form [{value: string, label: string}, ...] or [val, val, ...]
+ * @prop placeholder, text to display when control is empty
+ * @prop itemRenderer, optional custom itemRenderer, a function that receives (item, itemProps)
+ */
+@hoistComponent()
+export class AsyncComboField extends BaseDropdownField {
+    @observable.ref @setter items = [];
+
+    static defaultProps = {
+        placeholder: 'Search'
+    }
+
+    delegateProps = ['className', 'style', 'placeholder', 'disabled'];
+
+    constructor(props) {
+        super(props);
+        this.addAutoRun(() => this.onQueryChange(), {delay: 100});
+    }
+
+    render() {
+        const {style, width, itemRenderer, disabled} = this.props;
+
+        const options = this.normalizeOptions(this.items),
+            value = this.renderValue;
+
+        return suggest({
+            popoverProps: {popoverClassName: Classes.MINIMAL},
+            $items: options,
+            onItemSelect: this.onItemSelect,
+            itemRenderer: itemRenderer || this.defaultItemRenderer,
```

Comment:
reading an overridden itemRenderer from props can probably be built into the shared getter